File: http/http-client-annotation-processor/src/main/java/ru/tinkoff/kora/http/client/annotation/processor/HttpClientUtils.java
package ru.tinkoff.kora.http.client.annotation.processor;

import ru.tinkoff.kora.annotation.processor.common.CommonUtils;

import javax.lang.model.element.TypeElement;

public class HttpClientUtils {
    public static String clientName(TypeElement httpClientType) {
        return CommonUtils.getOuterClassesAsPrefix(httpClientType) + httpClientType.getSimpleName() + "_" + "ClientImpl";
    }

    public static String configName(TypeElement httpClientType) {
        return CommonUtils.getOuterClassesAsPrefix(httpClientType) + httpClientType.getSimpleName() + "_" + "Config";
    }

    public static String moduleName(TypeElement httpClientType) {
        return CommonUtils.getOuterClassesAsPrefix(httpClientType) + httpClientType.getSimpleName() + "_" + "Module";
    }
}

File: http/http-client-symbol-processor/src/main/kotlin/ru/tinkoff/kora/http/client/symbol/processor/HttpClientUtils.kt
package ru.tinkoff.kora.http.client.symbol.processor

import com.google.devtools.ksp.symbol.KSClassDeclaration
import com.squareup.kotlinpoet.ClassName
import ru.tinkoff.kora.ksp.common.getOuterClassesAsPrefix

fun KSClassDeclaration.clientName() = getOuterClassesAsPrefix() + simpleName.getShortName() + "_ClientImpl"

fun KSClassDeclaration.configName() = getOuterClassesAsPrefix() + simpleName.getShortName() + "_Config"

fun KSClassDeclaration.moduleName() = getOuterClassesAsPrefix() + simpleName.getShortName() + "_Module"

val httpClientAnnotation = ClassName("ru.tinkoff.kora.http.client.common.annotation", "HttpClient")
val httpClient = ClassName("ru.tinkoff.kora.http.client.common", "HttpClient")
val httpClientRequestMapper = ClassName("ru.tinkoff.kora.http.client.common.request", "HttpClientRequestMapper")
val httpClientResponseMapper = ClassName("ru.tinkoff.kora.http.client.common.response", "HttpClientResponseMapper")
